File: flutter_inappwebview_platform_interface/lib/src/types/launching_external_uri_scheme_request.dart
import 'package:flutter_inappwebview_internal_annotations/flutter_inappwebview_internal_annotations.dart';

import '../web_uri.dart';
import 'enum_method.dart';

part 'launching_external_uri_scheme_request.g.dart';

///Class that represents the request used by the [PlatformWebViewCreationParams.onLaunchingExternalUriScheme] event.
@ExchangeableObject()
class LaunchingExternalUriSchemeRequest_ {
  ///The URI with the external URI scheme to be launched.
  WebUri uri;

  ///The origin initiating the external URI scheme launch.
  WebUri? initiatingOrigin;

  ///Whether the external URI scheme request was initiated through a user gesture.
  bool? isUserInitiated;

  LaunchingExternalUriSchemeRequest_({
    required this.uri,
    this.initiatingOrigin,
    this.isUserInitiated,
  });
}

File: flutter_inappwebview_windows/windows/types/launching_external_uri_scheme_request.cpp
#include "launching_external_uri_scheme_request.h"

namespace flutter_inappwebview_plugin
{
  LaunchingExternalUriSchemeRequest::LaunchingExternalUriSchemeRequest(const std::string& uri, const std::optional<std::string>& initiatingOrigin,
    const std::optional<bool>& isUserInitiated)
    : uri(uri), initiatingOrigin(initiatingOrigin), isUserInitiated(isUserInitiated)
  {}

  flutter::EncodableMap LaunchingExternalUriSchemeRequest::toEncodableMap() const
  {
    return flutter::EncodableMap{
      {"uri", make_fl_value(uri)},
      {"initiatingOrigin", make_fl_value(initiatingOrigin)},
      {"isUserInitiated", make_fl_value(isUserInitiated)}
    };
  }
}